src/fs/data/private/spiralratchet.ts

```diff
@@ -11,13 +11,13 @@ interface SpiralRatchet {
 }
 
 
-export async function setup(ctx: EncryptionContext): Promise<SpiralRatchet> {
+export async function setup(ctx: EncryptionContext, seed?: ArrayBuffer, ffSmall?: number, ffMedium?: number): Promise<SpiralRatchet> {
   const { webcrypto, crypto } = ctx
   let [mediumSkip, smallSkip] = crypto.getRandomValues(new Uint32Array(2))
-  mediumSkip = mediumSkip % 255
-  smallSkip = smallSkip % 255
+  mediumSkip = ffMedium === undefined ? mediumSkip % 256 : ffMedium
+  smallSkip = ffSmall === undefined ? smallSkip % 256 : ffSmall
```

Comment:
This doesn't adhere to the spec here. The spec says:
```hs
setup :: SpiralRatchet
setup = do
  largePre   <- random -- ratchet seed
  mediumSkip <- randomBetween 0 254
  smallSkip  <- randomBetween 0 254
```
So, `randomBetween 0 254`. Is the spec wrong? (To be fair, I should've mentioned that I was confused by that in the spec.

If so, then we can simplify this code:
```ts
let mediumSkip = crypto.getRandomValues(new Uint8Array(2))
```
No need to `% 256` if they're bytes then :+1: 

Very minor nitpick: I like to make all defined-ness checks `something == null`. It checks both whether it's `undefined` or `null`.